extensions/csv/src/test/java/io/deephaven/csv/CsvTest.java

```diff
@@ -1,529 +1,638 @@
-                .header(ColumnHeader.ofString("Column3")));
+    private static final String WHITESPACE_INSIDE_AND_OUTSIDE = ("" +
+            "Sym,Type,Price,SecurityId\n" +
+            "=GOOG=, = Dividend=, = 0.25=, = 200=\n" +
+            "=T=, = Dividend=, = 0.15=, = 300=\n" +
+            "= Z=, = Dividend=, = 0.18=, = 500=\n")
+                    .replace('=', '"'); // using = instead of quotes in this code for readability
+
+    @Test
+    public void whitespaceInsideAndOutsideDefault() {
+        final Table expected = TableTools.newTable(
+                TableTools.col("Sym", "GOOG", "T", " Z"),
+                TableTools.col("Type", " Dividend", " Dividend", " Dividend"),
+                TableTools.col("Price", 0.25, 0.15, 0.18),
+                TableTools.col("SecurityId", (short) 200, (short) 300, (short) 500)
+        );
+        final CsvSpecs specs = CsvSpecs.csv();
+        invokeTest(WHITESPACE_INSIDE_AND_OUTSIDE, specs, expected);
     }
 
-    private static NewTable populateLanguageExample(ColumnHeaders3<String, String, String> header) {
-        return header
-                .row("C", "Dennis Ritchie", "Compiled")
-                .row("C++", "Bjarne Stroustrup", "Compiled")
-                .row("Fortran", "John Backus", "Compiled")
-                .row("Java", "James Gosling", "Both")
-                .row("JavaScript", "Brendan Eich", "Interpreted")
-                .row("MATLAB", "Cleve Moler", "Interpreted")
-                .row("Pascal", "Niklas Wirth", "Compiled")
-                .row("Python", "Guido van Rossum", "Interpreted")
-                .newTable();
+    @Test
+    public void whitespaceInsideAndOutsideTrim() {
+        final Table expected = TableTools.newTable(
+                TableTools.col("Sym", "GOOG", "T", "Z"),
+                TableTools.col("Type", "Dividend", "Dividend", "Dividend"),
+                TableTools.col("Price", 0.25, 0.15, 0.18),
+                TableTools.col("SecurityId", (short) 200, (short) 300, (short) 500)
+        );
+        final CsvSpecs specs = CsvSpecs.builder().trim(true).build();
+        invokeTest(WHITESPACE_INSIDE_AND_OUTSIDE, specs, expected);
+    }
+//
+    @Test
+    public void allNulls() {
+        final String input = "" +
+                "Long\n" +
+                "\n" +
+                "\n" +
+                "\n" +
+                "\n" +
+                "\n";
+        final Table expected = TableTools.newTable(
+                new ColumnHolder<>("Long", Long.class, null, false, null, null, null, null, null)
+        );
+        invokeTest(input,
+                CsvSpecs.builder().putParserForName("Long", Parsers.LONG).build(),
+                expected);
     }
 
-    private final String name;
-    private final String resourceName;
-    private final CsvSpecs specs;
-    private final NewTable expected;
 
-    public CsvTest(String name, String resourceName, CsvSpecs specs, NewTable expected) {
```

Comment:
Is there a reason that the parameterized CsvTest won't work?

Reply:
As I understand it, the way parametrized tests are supposed to be used is you have a test, and then you parameterize it. For example, you might write a test and then run it with size 10, 1000, 1 million, 1 billion.

Instead, what this code did was to have like 20 completely different tests, wrapped them in objects, made an array of those objects, and then considered those objects to be "parameters" to a single meta-test.

Of course you can do this, but I did not see what value it was adding, and in fact it was bothering me a lot because it was taking away the convenience of being able to set a breakpoint, right click on a test, and re-run a specific test (without re-running all of them) from the IDE. I was frequently commenting out tests from the array and recompiling the code and that was getting tiresome.